GoogleForCreators/web-stories-wp
# Video: Display captions using `<amp-story-captions>`

## Context



`<amp-story-captions>` allows us to improve the appearance of video captions on the page.

## Summary



This PR changes captions rendering to us the new `<amp-story-captions>` element (not in prod yet).

**Note:** For testing you might need to opt into the nightly/beta channel at https://cdn.ampproject.org/experiments.html first.

## Relevant Technical Choices



- Temporarily patches the AMP plugin since `<amp-story-captions>` is not yet in prod.

## To-do



- [x] Improve default styling

## User-facing changes



| **Before**  | **After** |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/841956/133491294-294974cb-77f7-4b44-9e7c-a4ca0c6c14f5.mov" muted />  | <video src="https://user-images.githubusercontent.com/841956/134395859-f858e4c8-c7eb-452f-a882-2eefb2a3157d.mov" muted />  |

Example of customized captions styling [using a WordPress plugin](https://gist.github.com/swissspidy/a0399febc85544d4e4992db3a7266f84):

![Screenshot 2021-09-15 at 20 40 05](https://user-images.githubusercontent.com/841956/133491119-366d53e6-8a1c-40b3-8918-b9064b02c6c2.png)

## Testing Instructions






This PR can be tested by following these steps:

1. Enable the video captions experiment
2. Create a story with 1 or more videos
3. Upload captions for the video(s)
4. Add some other elements on layers above the video(s)
5. Preview story and verify that captions are displayed correctly and on top of all other elements on the page and within the safe zone

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



See #5315
Fixes #9154
